stingray/deadtime/fad.py

```diff
@@ -7,63 +7,34 @@
 from scipy.interpolate import UnivariateSpline
 from astropy import log
 from astropy.table import Table
-from ..crossspectrum import AveragedCrossspectrum, normalize_crossspectrum
+
+from stingray.lightcurve import Lightcurve
+from ..crossspectrum import AveragedCrossspectrum, show_progress, get_flux_generator
 from ..powerspectrum import AveragedPowerspectrum
+from ..fourier import normalize_periodograms, fft, fftfreq, positive_fft_bins
 
 from ..gti import cross_two_gtis, bin_intervals_from_gtis
 
 
-__all__ = ["calculate_FAD_correction", "get_periodograms_from_FAD_results"]
-
-
-def _get_fourier_intv(lc, start_ind, end_ind):
-    """Calculate the Fourier transform of a light curve chunk.
-
-    Parameters
-    ----------
-    lc : a :class:`Lightcurve` object
-        Input light curve
-
-    start_ind : int
-        Start index of the light curve chunk
-
-    end_ind : int
-        End index of the light curve chunk
-
-    Returns
-    -------
-    freq : array of floats
-        Frequencies of the Fourier transform
-
-    fft : array of complex numbers
-        The Fourier transform
-
-    nph : int
-        Number of photons in the interval of the light curve
-
-    nbins : int
-        Number of bins in the light curve segment
-
-    meancounts : float
-        The mean counts/bin in the light curve
-    """
-    time = lc.time[start_ind:end_ind]
-    counts = lc.counts[start_ind:end_ind]
-
-    fourier = scipy.fft.fft(counts)
-
-    freq = scipy.fft.fftfreq(len(time), lc.dt)
-    good = freq > 0
-
-    nbins = time.size
-    return freq[good], fourier[good], np.sum(counts), nbins
-
-
-def calculate_FAD_correction(lc1, lc2, segment_size, norm="none", gti=None,
-                             plot=False, ax=None, smoothing_alg='gauss',
-                             smoothing_length=None, verbose=False,
-                             tolerance=0.05, strict=False, all_leahy=False,
-                             output_file=None, return_objects=False):
+__all__ = ["calculate_FAD_correction", "get_periodograms_from_FAD_results", "FAD"]
+
+
+def FAD(
+    data1,
+    data2,
+    segment_size,
+    dt=None,
+    norm="abs",
+    plot=False,
+    ax=None,
+    smoothing_alg='gauss',
+    smoothing_length=None,
+    verbose=False,
+    tolerance=0.05,
+    strict=False,
+    output_file=None,
+    return_objects=False
+):
```

Comment:
Is this the linter again? Do we really want to force parameters to each have their own line? That's not in our coding standards, and I personally don't find that it makes the code more readable.